### ISSUE TITLE
Add types to the arguments that are passed to ws.connect

### ISSUE THREAD

**Issue by ramya-rao-a** [amqp/rhea#209]
[ws.connect](https://github.com/amqp/rhea/blob/70e51166e822d51fc463674729af12970a0fb2ba/typings/ws.d.ts#L4) is takes in a parameter `Impl` which is meant to be a constructor for the WebSocket.

@grs, @amarzavery Any reason why this was not typed in the initial effort to get types for the rhea library?

If there were no strong reasons for skipping the type of `Impl`, then we would like to contribute the types as below:

```typescript
/**
 * Describes the required shape of WebSocket instances.
 * @interface WebSocketInstance
 */
export interface WebSocketInstance {
  send: Function;
  onmessage: Function | null;
  onopen: Function | null;
  onclose: Function | null;
  onerror: Function | null;
}

/**
 * Describes the required shape of WebSocket constructors.
 * @interface WebSocketImpl
 */
export interface WebSocketImpl {
  new(url: string, protocols?: string | string[]): WebSocketInstance;
}
```

cc @bterlson, @AlexGhiondea

**Comment by amarzavery** [comment]
Websockets was one of the last things on my to-do list when I started providing type definitions. Since I didn't have enough information about it, I left it to be updated later on. If @grs thinks that the above definition matches the implementation then please send a PR to update type definition.

**Comment by ramya-rao-a** [comment]
@ShivangiReja, @bterlson,

Would the above plan still holds true based on your current plans for WebSockets?

**Comment by bterlson** [comment]
I think so, with the exception of the third `options` parameter that @shivangireja added.

We were also considering making the option type generic so that TypeScript can infer the options type so e.g. people can bring whatever WS implementation they want and the options will respect that.

**Comment by ramya-rao-a** [comment]
Closing this issue as we have implemented required changes in #214
